docs/policies/rate-limit.md

```diff
@@ -78,20 +78,29 @@ requests, every 1 minute for each unique IP address.
 ## Using a custom function
 
 You can create a rate-limit bucket based on any property of a request using a
-custom function that returns a string (that is the identified used by the
-limiting system).
+custom function that returns a `CustomRateLimitPolicyOptions` object (which provides the identifier used by the limiting system).
 
-This example would create a unique rate-limiting function based on the `customerId`
-parameter in routes (note it’s important that a policy like this is applied to a
-route that has a `/:customerId` parameter).
+The `CustomRateLimitPolicyOptions` object can be used to override the `timeWindowMinutes` & `requestsAllowed` options.
+
+This example would create a unique rate-limiting function based on the `customerId` parameter in routes (note it’s important that a policy like this is applied to a route that has a `/:customerId` parameter).
 
 ```ts
 //module - ./modules/rate-limiter.ts
 
-import { ZuploRequest } from "@zuplo/runtime"
-
-export function rateLimitKey(request:ZuploRequest) {
-  return request.params.customerId;
+import { CustomRateLimitPolicyOptions, ZuploRequest } from "@zuplo/runtime";
+
+export function rateLimitKey(
+  request: ZuploRequest
+): CustomRateLimitPolicyOptions {
+  if (request.params.customerId === "43567890") {
+    // Override timeWindowMinutes & requestsAllowed
+    return {
```

Comment:
Nice, what happens if I keep giving random numbers for the requestsAllowed for the same key?

Reply:
It will vary as new cloudflare instances get created and the bucket expires. For example, if everything is processed by only one instance then the first request is the one that initializes the rate limit bucked using the provided options. That will stay that way until the time limit expires. 

Once more instances get spun up then their rate limit bucket for that customer id will be configured according to the options provided and the previously mentioned logic will apply.

Reply:
basically, don't mess about with the limits, try to keep it consistent for a given key or you'll see weird behavior. Fine to change occasionally though.

Reply:
Yeah, the idea is for this to allow a user to customize behavior based on keys. Not sure if they'll want to have those limit be dynamic on the same key but that is possible too.